### PR TITLE
request 2

### PR DESCRIPTION
**Pull request by Luan-N**
modified apppanel, model, and view. Professor made changes to the Utilities file so it was also updated. Unsure how to implement safeframe to mvc. stoplight now works with the mvc except for the safeframe part.

### CODE REVIEW

File: src/mvc/Utilities.java [hunk 1]
```diff
@@ -33,8 +28,8 @@ public static void inform(String info) {
     // tells user a lot of info
     public static void inform(String[] items) {
         String helpString = "";
-        for (String item : items) {
-            helpString = helpString + "\n" + item;
+        for(int i = 0; i < items.length; i++) {
```
Comment (jonetiz) [inline src/mvc/Utilities.java:31]: this is prof's original code but was improved with the improved for loop

File: src/mvc/Utilities.java [hunk 2]
```diff
@@ -120,8 +117,8 @@ public static Model open(Model model) {
     // simple menu maker
     public static JMenu makeMenu(String name, String[] items, ActionListener handler) {
         JMenu result = new JMenu(name);
-        for (String s : items) {
-            JMenuItem item = new JMenuItem(s);
+        for(int i = 0; i < items.length; i++) {
```
Comment (jonetiz) [inline src/mvc/Utilities.java:121]: this is prof's original code but was improved with the improved for loop

File: src/mvc/AppPanel.java [hunk 3]
```diff
@@ -48,11 +60,67 @@ public void display(){
     }
 
     public void actionPerformed(ActionEvent e) {
+        String cmmd = e.getActionCommand();
+        System.out.println(cmmd);
+        try {
+            switch (cmmd) {
+                case "Save": {
+                    String fName = Utilities.getFileName((String) null, false);
```
Comment (jonetiz) [inline src/mvc/AppPanel.java]: Use Utilities.save() instead, it's safer and already implemented anyway.

File: src/mvc/AppPanel.java [hunk 4]
```diff
@@ -48,11 +60,67 @@ public void display(){
...
+                    ObjectOutputStream os = new ObjectOutputStream(new FileOutputStream(fName));
+                    os.writeObject(this.model);
+                    os.close();
+                    break;
+                }
+
+                case "Open": {
+
+                    if (Utilities.confirm("Are you sure? Unsaved changes will be lost!")) {
```
Comment (jonetiz) [inline src/mvc/AppPanel.java]: Use Utilities.open() instead, it's safer and already implemented anyway (ditto with Utilities.save())

**Review by jonetiz — CHANGES_REQUESTED** [review]
In utilities.java, we had improved for loops for `inform` and `makeMenu` functions, additionally Utilities already implements a `save` and `open` function which should be used on the AppPanel instead of implementing new functionality.

**Comment by Luan-N** [comment]
I’ve just made the changes you asked for and hit commit. Do I have to send another pull request?

**Pull request merged**